addons/account/models/account_invoiceable_line.py

```diff
@@ -0,0 +1,79 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import _, api, fields, models
+from odoo.tools import format_date
+
+
+class InvoiceableLineMixin(models.AbstractModel):
+    _name = 'account.invoiceable.line'
+    _description = 'Enables downpayment support for child records of account.invoiceable.order that can be down payments'
+
+    is_downpayment = fields.Boolean(string="Is a down payment",
+                                    help="Down payments are made when creating account moves from an order."
+                                         " They are not copied when duplicating an order.")
+
+    # To override
+    invoice_lines = fields.One2many('account.move.line')
+    display_type = fields.Selection([
+        ('line_section', "Section"),
+        ('line_note', "Note")], default=False, help="Technical field for UX purpose.")
+    product_id = fields.Many2one('product.product')
+    name = fields.Text(compute='_compute_name')
+    order_id = fields.Many2one('account.invoiceable.order')
+    qty_to_invoice = fields.Float()
+    price_unit = fields.Float()
+    tax_id = fields.Many2many('account.tax')
+
+    def _get_invoice_lines(self):
+        raise NotImplementedError # To override
+
+    def _valid_qty_to_invoice(self, final=False):
+        raise NotImplementedError # To override
+
+    def _prepare_invoice_line(self, move=False, **optional_values):
+        raise NotImplementedError # To override
+
+    def _get_description(self):
+        raise NotImplementedError # To override
```

Comment:
All those hook functions need a proper docstring to explain when they are called and to which purpose.